File: frontend/src/types/SaveProfilePayload.ts
import type { EmailInput } from "./EmailInput";

export type SaveProfilePayload = {
  email: EmailInput;
  firstName: string;
  middleName?: string;
  lastName: string;
  nickname?: string;
  birthdate?: Date;
  gender?: string;
  locale: string;
  timeZone?: string;
  picture?: string;
  profile?: string;
  website?: string;
};

File: backend/src/Logitar.Demo.Web/Models/Account/SaveProfilePayload.cs
using Logitar.Portal.Contracts.Users.Contact;

namespace Logitar.Demo.Web.Models.Account;

public record SaveProfilePayload
{
  public EmailInput Email { get; set; } = new();

  public string FirstName { get; set; } = string.Empty;
  public string? MiddleName { get; set; }
  public string LastName { get; set; } = string.Empty;
  public string? Nickname { get; set; }

  public string Locale { get; set; } = string.Empty;
  public string? TimeZone { get; set; }
}
